Dart/src/com/jetbrains/lang/dart/ide/surroundWith/expression/DartWithBracketsExpressionSurrounder.java

```diff
@@ -0,0 +1,33 @@
+package com.jetbrains.lang.dart.ide.surroundWith.expression;
+
+import com.intellij.psi.PsiElement;
+import com.jetbrains.lang.dart.psi.DartNamedArgument;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DartWithBracketsExpressionSurrounder extends DartWithExpressionSurrounder {
+
+  public boolean isApplicable(@NotNull PsiElement[] elements) {
+    // Limit this to named arguments; the intent is to convert a Flutter child: param to children:, which may involve creating red code.
+    return super.isApplicable(elements) && elements[0].getParent() instanceof DartNamedArgument;
+  }
+
+  @Override
+  public String getTemplateDescription() {
+    return "[expr]";
+  }
+
+  @Override
+  protected String getTemplateText(PsiElement expr) {
+    String text = expr.getText();
+    int nlIndex = text.lastIndexOf('\n');
+    if (nlIndex < 0) {
+      return "[" + expr.getText() + "]";
+    }
+    Matcher matcher = Pattern.compile("\\n", Pattern.MULTILINE).matcher(text);
+    String newText = matcher.replaceAll("\n  ");
```

Comment:
Indents handling seems to be not needed, IDE auto-reformats the code after the action. I [simplified](https://github.com/JetBrains/intellij-plugins/commit/60bdb59e08d2b4e8584a3d8bd0cd82022a998dd1) this and tests still pass. Let me know if I missed something.